bundles/framework/admin-announcements/view/DeleteButton.jsx

```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import "antd/dist/antd.css";
+import PropTypes from 'prop-types';
+import { Button } from "antd";
+
+//TODO: Separate component for deleting announcements. Could not get this to work properly yet.
+
+const DeleteButton = ({ controller, deleteAnnouncement }) => {
+    return (
+    <Confirm
+        title={<React.Fragment>
+            <div>
+                <Message messageKey='messages.confirmDeleteAnnouncement'/>
+            </div>
+            { layerCountInGroup > 0 &&
+                <DeleteLayersCheckbox checked={deleteLayers} onChange={ evt => controller.setDeleteLayers(evt.target.checked)}>
```

Comment:
I think this file can be just removed. It doesn't appear to be used anywhere. In addition it references announcements but also deleting layers? Some DeleteButton component could be added to src/react/components but the confirmation text and click callback should be given as params.

Reply:
This was the unfinished file that was supposed to become a component. I will stash it on my own computer along with this comment for future development.